packages/webapp/client/js/core.js

```diff
@@ -14,3 +27,80 @@ const jsElements = document.getElementsByClassName('defra-js-show')
 Array.prototype.forEach.call(jsElements, function (element) {
   element.removeAttribute('hidden')
 })
+
+const acceptButton = document.querySelector('.js-cookies-button-accept')
+const rejectButton = document.querySelector('.js-cookies-button-reject')
+const acceptedBanner = document.querySelector('.js-cookies-accepted')
+const rejectedBanner = document.querySelector('.js-cookies-rejected')
+const questionBanner = document.querySelector('.js-question-banner')
+const cookieBanner = document.querySelector('.js-cookies-banner')
+const cookieBannerContainer = document.querySelector('.js-cookie-banner-container')
+const cookiePrefsPage = document.querySelector('.defra-js')
+const cookiePageBanner = document.querySelector('.js-cookie-page-banner')
+const cookieSeenBanner = 'seen_cookie_message'
+const cookieSeenBannerExpiry = 30
+const cookieUserPreference = 'set_cookie_usage'
+const cookieUserPreferenceExpiry = 30
+
+const savePreference = accepted => {
+  const prefs = {
+    analytics: accepted ? 'on' : 'off'
+  }
+  window.bng.utils.setCookie(cookieUserPreference, JSON.stringify(prefs), cookieUserPreferenceExpiry)
+  cookiePageBanner.style.display = 'block'
+  cookiePageBanner.setAttribute('tabindex', '-1')
+  cookiePageBanner.focus()
+}
+
+if (cookiePrefsPage) {
+  const saveButton = document.querySelector('#cookies-save')
+  cookiePageBanner.style.display = 'none'
+  saveButton?.addEventListener('click', function (event) {
```

Comment:
Also when testing if setting analytics preferences on the cookies page it isn't setting the banner seen cookie, so banner still shows after setting cookies, needs to be hidden I would have thought